### PR TITLE
Generalize discovery API overrides to support regex fields

### PR DESCRIPTION
**Pull request by tcoffee-google**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/discoveries/cloudmonitoring.v2beta2.json.overrides [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
   "java": {
     "methods": {
-      "cloudmonitoring.metricDescriptors.list": {
-        "requestBodyType": null
-      },
-      "cloudmonitoring.timeseries.list": {
-        "requestBodyType": null
-      },
-      "cloudmonitoring.timeseriesDescriptors.list": {
+      ".*list": {
```
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/cloudmonitoring.v2beta2.json.overrides]: While unlikely, it's possible that they add another list method that is actually correct.
Should we do `cloudmonitoring.(metricDescriptors|timeseries|timeseriesDescriptors).list" instead?

File: src/test/java/com/google/api/codegen/testdata/discoveries/translate.v2.json.overrides [hunk 2]
```diff
@@ -1,17 +1,7 @@
...
-          "q": {
-            "type": {
-              "kind": "TYPE_STRING",
-              "isArray": false
-            }
-          }
-        }
-      },
-      "language.translations.list": {
+      "language.(detections|translations).list": {
```
Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/translate.v2.json.overrides]: Do the periods here correspond to any character or do they need to be escaped?

File: src/test/java/com/google/api/codegen/testdata/discoveries/cloudmonitoring.v2beta2.json.overrides (hunk 1 above)
Comment (tcoffee-google) [inline src/test/java/com/google/api/codegen/testdata/discoveries/cloudmonitoring.v2beta2.json.overrides]: I considered it, but thought it more likely such a hypothetical method would follow their existing pattern. Seems like compile check would catch it either way.

File: src/test/java/com/google/api/codegen/testdata/discoveries/translate.v2.json.overrides (hunk 2 above)
Comment (tcoffee-google) [inline src/test/java/com/google/api/codegen/testdata/discoveries/translate.v2.json.overrides]: They're any character, but the method ID format means they must be periods.

**Comment by saicheems** [comment]
lgtm

**Pull request merged**